binderhub/builder.py

```diff
@@ -425,33 +425,72 @@ async def get(self, provider_prefix, _unescaped_spec):
             ref_url=self.ref_url,
         )
 
-        self.build = build = BuildClass(
-            q=q,
-            # api object can be None if we are using FakeBuild
-            api=self.settings.get("kubernetes_client"),
-            name=build_name,
-            namespace=self.settings["build_namespace"],
-            repo_url=repo_url,
-            ref=ref,
-            image_name=image_name,
-            push_secret=push_secret,
-            build_image=self.settings["build_image"],
-            memory_limit=self.settings["build_memory_limit"],
-            memory_request=self.settings["build_memory_request"],
-            docker_host=self.settings["build_docker_host"],
-            node_selector=self.settings["build_node_selector"],
-            appendix=appendix,
-            log_tail_lines=self.settings["log_tail_lines"],
-            git_credentials=provider.git_credentials,
-            sticky_builds=self.settings["sticky_builds"],
-        )
+        if issubclass(BuildClass, Build):
+            # Deprecated
```

Comment:
Maybe this can be described a bit or a link to the PR description as a reference to an explanation on what is going on?

From the PR description:

> The `build.py` changes look very large, but it's mostly rearranging the existing code:
> 
> * Common parameters and methods are moved out of `Build` and into the Traitlets based `BuildExecutor` base class (suggestions for alternative names welcome- if we want backwards compatibility we have to keep the generically named `Build` class as the Kubernetes builder)
> * K8S specific methods are moved from `Build` to the Traitlets based `KubernetesBuildExecutor`
> * The original `Build` class is now just a wrapper class around `KubernetesBuildExecutor`. The `Build.__init__` function signature is unchanged, so hopefully if someone's subclassed it their class should still work with binderHub.
> 
> `builder.py` now checks whether the BuildClass is the old existing `Build` or the newer `BuildExecutor` based classes. If it's the former the behaviour is unchanged. If it's the latter then parameters should be set using Traitlets in a config file or similar, the current BinderHub command-line args will be ignored.

Reply:
```suggestion
        if issubclass(BuildClass, Build):
            # Deprecated, see docstring of the Build class for more details
```